File: src/main/java/incom/hyogyu/gui/GUIManager.java
package incom.hyogyu.gui;

import incom.hyogyu.gui.games.DodgeMasterPanel;
import incom.hyogyu.gui.games.SnakeGamePanel;

import javax.swing.*;
import java.awt.*;

public class GUIManager {
    private final JFrame frame;
    private final CardLayout cardLayout;
    private final JPanel mainPanel;
    private final SnakeGamePanel snakeGame;
    private final DodgeMasterPanel dodgeMaster;
    private final MainMenuPanel mainMenu;

    public GUIManager() {
        frame = new JFrame("[INCOM] 동아리 박람회 게임");
        frame.setSize(1366, 768);
        frame.setResizable(false);

        mainPanel = new JPanel();
        cardLayout = new CardLayout();
        mainPanel.setLayout(cardLayout);

        mainMenu = new MainMenuPanel(this);
        snakeGame = new SnakeGamePanel(this);
        dodgeMaster = new DodgeMasterPanel(this);

        mainPanel.add(mainMenu, "MainMenu");
        mainPanel.add(snakeGame, "SnakeGame");
        mainPanel.add(dodgeMaster, "DodgeMaster");

        frame.add(mainPanel);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.pack();
        frame.setLocationRelativeTo(null);
        frame.setVisible(true);
    }

    public void switchTo(String panelName) {
        snakeGame.stopGame();
        dodgeMaster.stopGame();

        cardLayout.show(mainPanel, panelName);
        if (panelName.equals("SnakeGame")) {
            snakeGame.startGame();
        } else if (panelName.equals("DodgeMaster")) {
            dodgeMaster.startGame();
        }
    }

}

File: pythonPorting/gui_manager.py
from PyQt6.QtWidgets import QWidget, QStackedWidget, QVBoxLayout
from main_menu_panel import MainMenuPanel
from snake_game_panel import SnakeGamePanel
from dodge_master_panel import DodgeMasterPanel

class GUIManager(QWidget):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setWindowTitle("[INCOM] 동아리 박람회 게임")
        self.setFixedSize(1366, 768)
        
        self.stackedWidget = QStackedWidget()
        
        self.mainMenu = MainMenuPanel(self)
        self.snakeGame = SnakeGamePanel(self)
        self.dodgeMaster = DodgeMasterPanel(self)
        
        self.stackedWidget.addWidget(self.mainMenu)
        self.stackedWidget.addWidget(self.snakeGame)
        self.stackedWidget.addWidget(self.dodgeMaster)
        
        layout = QVBoxLayout()
        layout.addWidget(self.stackedWidget)
        self.setLayout(layout)
    
    def switchTo(self, panelName: str):
        self.snakeGame.stopGame()
        self.dodgeMaster.stopGame()
        
        if panelName == "MainMenu":
            self.stackedWidget.setCurrentWidget(self.mainMenu)
        elif panelName == "SnakeGame":
            self.stackedWidget.setCurrentWidget(self.snakeGame)
            self.snakeGame.startGame()
        elif panelName == "DodgeMaster":
            self.stackedWidget.setCurrentWidget(self.dodgeMaster)
            self.dodgeMaster.startGame()